src/graph.rs

```diff
@@ -883,6 +883,12 @@ impl<N, E, Ty, Ix> Graph<N, E, Ty, Ix>
         &self.edges
     }
 
+    /// Return the internal arrays, consuming the graph.
+    pub fn destructure(self) -> (Vec<Node<N, Ix>>, Vec<Edge<E, Ix>>)
```

Comment:
I suppose both variants would work for my use case, the consuming one just felt more natural (and, apparently, more general).